### ISSUE TITLE
Two arithmetic overflow found in DynamicImage::thumbnail_exact

### ISSUE THREAD

**Issue by thdusdl1219** [image-rs/image#2015]
I found two arithmetic overflow cases in the latest version.

### Case 1

This happens in 

https://github.com/image-rs/image/blob/f1e34f365133a33a27dac5bc59c93d1cfb34a37e/src/imageops/sample.rs#L579

### Case 2

This happens in 

https://github.com/image-rs/image/blob/f1e34f365133a33a27dac5bc59c93d1cfb34a37e/src/imageops/sample.rs#L586

## Expected

Assert or size check before subtraction or use the checked operation instead.

### Case 1
`thread 'main' panicked at 'attempt to subtract with overflow', .cargo/registry/src/index.crates.io-6f17d22bba15001f/image-0.24.7/./src/imageops/sample.rs:579:54`

### Case 2
`thread 'main' panicked at 'attempt to subtract with overflow', .cargo/registry/src/index.crates.io-6f17d22bba15001f/image-0.24.7/./src/imageops/sample.rs:586:54`

### Case 1

```
    let dyimg = DynamicImage::default();
    dyimg.thumbnail_exact(0, 1);
```

### Case2

```
    let dyimg = DynamicImage::new_rgb8(0, 1);
    dyimg.thumbnail_exact(1, 1);
```

**Comment by fintelia** [comment]
Pull requests would be appreciated!

**Comment by fintelia** [comment]
Fixed by #2041
